libraries/AP_Airspeed/AP_Airspeed_SST_ND.cpp

```diff
@@ -0,0 +1,306 @@
+
+    if (_dev_model != DevModel::SST_ND){
+        diff_press_inH2O = (dp_raw * _current_range_val) / margin;
+    }
+    else{
+        diff_press_inH2O = (float)(_current_range_val *
+                                    (dp_raw - 8388607.5f) /
+                                    15099493.5f);
+    }
+    float press  = diff_press_inH2O * inH20_to_Pa;
+    return press;
+}
+
+/*
+  convert raw temperature to temperature in degrees C
+ */
+float AP_Airspeed_SST_ND::_get_temperature(int8_t dT_int, int8_t dT_frac) const
+{
+    float temp  = dT_int + dT_frac/256.0;
+    return temp;
+}
+
+// read the values from the sensor
+void AP_Airspeed_SST_ND::_collect()
+{
+    uint8_t data[6]; //2 bytes for pressure and 2 for temperature
+    _dev->get_semaphore()->take_blocking();
+    if (!_dev->read(data, sizeof(data))) {
+        return;
+    }
+    _dev->get_semaphore()->give();
+
+    uint32_t dp_raw = 0x00;
+    dp_raw = (data[1] << 16) | (data[2] << 8) | data[3];
+
+    float press  = _get_pressure(dp_raw);
+    float temp  = _get_temperature(data[4], data[5]);
+
+    WITH_SEMAPHORE(sem);
+
+    _press_sum += press ;
+    _temp_sum += temp;
+    _press_count += 1;
+    _temp_count += 1;
+
+    _last_sample_time_ms = AP_HAL::millis();
+}
+
+bool AP_Airspeed_SST_ND::range_change_needed(float last_pressure) {
+    if(last_pressure > 0.8*_current_range_val*inH20_to_Pa){ // if above 85% of range, go to the next
+        if(_range_setting > 0){
+            _range_setting -= 1;
+            return true;
+        }
+    }else if(last_pressure < 0.25*_current_range_val*inH20_to_Pa){ // if below 15% of range, go to the next
+        if(_range_setting < _available_ranges - 1){
+            _range_setting += 1;
+            return true;
+        } 
+    }
+    return false;
+}
+
+// set the range of the sensor
+void AP_Airspeed_SST_ND::update_range()
+{
+    switch(_dev_model){
+        case DevModel::ND210:
+            _current_range_val = nd210_range[_range_setting];
+            break;
+        case DevModel::ND005D:
+            _current_range_val = nd005d_range[_range_setting];
+            break;
+        default:
+            _current_range_val = 0.0f;
+            //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"No specific device detected/not supported\n");
+    }
+    config_setting[0] = (config_setting[0] & 0xF0) + (0b0111 - _range_setting);
+    WITH_SEMAPHORE(_dev->get_semaphore());
+    //_dev->transfer(config_setting, 2, nullptr,0);
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Range changed to %d: %.2f inH2O\n", _range_setting, _current_range_val);
+    hal.scheduler->delay(2); // wait for the sensor to change range
+}
+
+// get the differential pressure in Pascals
+bool AP_Airspeed_SST_ND::get_differential_pressure(float &pressure)
+{
+    WITH_SEMAPHORE(sem);
+
+    if ((AP_HAL::millis() - _last_sample_time_ms) > 100 || _press_count == 0) {
+        return false;
+    }
+
+    _pressure = _press_sum / _press_count;
+    _press_count = 0;
+    _press_sum = 0;
```

Comment:
```suggestion
    if (_press_count > 0) {
        _pressure = _press_sum / _press_count;
        _press_count = 0;
        _press_sum = 0;
    }
```

similarly for temperature